src/Compilers/CSharp/Portable/LanguageVersion.cs

```diff
@@ -65,18 +65,56 @@ public enum LanguageVersion
         CSharp6 = 6,
 
         /// <summary>
-        /// C# language version 7.
+        /// C# language version 7.0
         /// </summary>
+        /// <remarks>
+        /// <para>Features:</para>
+        /// <list type="bullet">
+        /// <item><description>Out variables</description></item>
+        /// <item><description>Pattern-matching</description></item>
+        /// <item><description>Tuples</description></item>
+        /// <item><description>Deconstruction</description></item>
+        /// <item><description>Discards</description></item>
+        /// <item><description>Local functions</description></item>
+        /// <item><description>Digit separators</description></item>
+        /// <item><description>Ref returns and locals</description></item>
+        /// <item><description>Generalized async return types</description></item>
+        /// <item><description>More expression-bodied members</description></item>
+        /// <item><description>Throw expressions</description></item>
+        /// </list>
+        /// </remarks>
         CSharp7 = 7,
 
         /// <summary>
         /// C# language version 7.1
         /// </summary>
+        /// <remarks>
+        /// <para>Features:</para>
+        /// <list type="bullet">
+        /// <item><description>Async Main</description></item>
+        /// <item><description>Default literal</description></item>
+        /// <item><description>Reference assemblies</description></item>
```

Comment:
Does it make sense to be pedantic about differentiating between what features are supported by each `LanguageVersion` and which version of Roslyn is required to use a feature?

I'm asking because creating reference assemblies using a recent version of Roslyn works even with e.g. `LanguageVersion.CSharp1`. For example, the following code correctly creates a reference assembly:

```c#
string code = @"
public class C
{
    private static int i = 42;
    public static void Main()
    {
        System.Console.WriteLine(i);
    }
}";

var tree = SyntaxFactory.ParseSyntaxTree(code, new CSharpParseOptions(LanguageVersion.CSharp1));

var compilation = CSharpCompilation.Create("a")
    .AddSyntaxTrees(tree)
    .AddReferences(MetadataReference.CreateFromFile(typeof(object).Assembly.Location));

using (var stream = File.OpenWrite("a.dll"))
{
    compilation.Emit(Stream.Null, metadataPEStream: stream);
}
```

Reply:
Good point, reference assemblies are not tied to a languages version.